crowbar_framework/spec/models/api/upgrade_spec.rb

```diff
@@ -519,6 +521,29 @@
         and_return([Node.find_by_name("testing.crowbar.com")])
       )
       allow_any_instance_of(Node).to receive(:upgraded?).and_return(true)
+      allow(Api::Upgrade).to receive(:prepare_all_compute_nodes).and_return(true)
+      allow(Api::Upgrade).to receive(:upgrade_all_compute_nodes).and_return(true)
+      allow(Api::Upgrade).to receive(:finalize_nodes_upgrade).and_return(true)
+      allow_any_instance_of(Crowbar::UpgradeStatus).to receive(:end_step).and_return(true)
+
+      expect(subject.class.nodes_without_delay).to be true
+    end
+
+    it "during preparation of compute nodes upgrade , detect that they are upgraded" do
```

Comment:
i'd rephrase this a bit like:
```ruby
it "detects that the nodes are already upgraded during preparation" do
```

or make the preparation a context like:
```ruby
context "during preparation of compute nodes upgrade" do
  it "detects that the nodes are already upgraded" do
```

...but having just one example in a context is also not preferable, but it sounds much better at least